draft-ietf-quic-transport.md

```diff
@@ -3838,143 +3838,135 @@ The details of loss detection and congestion control are described in
-ECN counts.  For example, if one each of an Initial, 0-RTT, Handshake, and 1-RTT
-QUIC packet are coalesced, the corresponding counts for the Initial and
-Handshake packet number space will be incremented by one and the counts for the
-application data packet number space will be increased by two.
+Even if an endpoint does not set an ECT codepoint on packets it sends, the
+endpoint MUST provide feedback about ECN codepoints it receives, if these are
+accessible.  Failing to report the ECN counts will cause the sender to disable
+use of ECN for packets to this receiver.
 
+On receiving a QUIC packet with an ECT(0), ECT(1) or CE codepoint, an
+ECN-enabled endpoint accesses the ECN codepoint from the enclosing IP packet and
+increases the corresponding ECT(0), ECT(1), or CE count, These ECN counts are
+included in subsequent ACK frames; see {{generating-acks}} and {{frame-ack}}.
 
-### ECN Validation {#ecn-validation}
+Each packet number space maintains separate acknowledgement state and separate
+ECN counts.  Coalesced packets (see {{packet-coalesce}}) carry several QUIC
+packets that share the same IP header.  The ECN count for the ECN codepoint
+received in the associated IP header are incremented once for each QUIC packet,
+not for the enclosing IP packet.
 
-It is possible for faulty network devices to corrupt or erroneously drop packets
-with ECN markings.  To provide robust connectivity in the presence of such
-devices, each endpoint independently validates ECN counts and disables ECN if
-errors are detected.
+For example, if one each of an Initial, 0-RTT, Handshake, and 1-RTT QUIC packet
+are coalesced, the corresponding counts for the Initial and Handshake packet
+number space will be incremented by one and the counts for the application data
+packet number space will be increased by two.
 
-Endpoints validate ECN for packets sent on each network path independently.  An
-endpoint thus validates ECN on new connection establishment, when switching to a
-server's preferred address, and on active connection migration to a new path.
-{{ecn-alg}} describes one possible algorithm for testing paths for ECN support.
+An ECN count is not incremented when the received IP packet does not results in
+a QUIC packets being processed.  A QUIC packet detected by a receiver as a
+duplicate also does not increase an ECN count; see {{security-ecn}} for relevant
+security concerns.
 
-Even if an endpoint does not use ECN markings on packets it transmits, the
-endpoint MUST provide feedback about ECN markings received from the peer if they
-are accessible.  Failing to report ECN counts will cause the peer to disable ECN
-marking.
 
+### ECN Validation {#ecn-validation}
 
-#### Sending ECN Markings
+It is possible for faulty network devices to corrupt or erroneously drop packets
+that set an ECN codepoint.  To provide robust connectivity in the presence of
+such devices, an endpoint validates the ECN counts for each network path  and
```

Comment:
This needs more explanation because you say first something about "drops" and the "validate ECN counts" but to check for drops you need to detect loss of ECT marked packet which is not really discussed here.

Reply:
This is introduction.  The explanation you seek follows.

Reply:
I just thought the two sentences don't fit together because the first talks about corrupt and drop but the second sentence only talk about validating counts and not about checking on loss. This is an editorial comment.